lib/ansible/plugins/callback/__init__.py

```diff
@@ -506,60 +506,93 @@ def v2_on_any(self, *args, **kwargs):
 
-        Customization note: For more information about the attributes and methods of the
-        TaskResult class, see lib/ansible/executor/task_result.py.
+        Customization note: Ansible uses only one console stdout
+        callback at a time. However, you can override this method in a
+        separate callback to add output to stdout or forward details
+        to a log, etc.
 
-        :param TaskResult result: An object that contains details about the task
+        :param result: An object that contains details about the task
+        :type result: TaskResult
 
         :return: None
         """
         host = result._host.get_name()
         self.runner_on_ok(host, result._result)
 
     def v2_runner_on_skipped(self, result: TaskResult) -> None:
-        """Get details about a skipped task and process them as required by the callback
-        (output, profiling, logging, notifications, etc.)
+        """Get details about a skipped task.
 
-        Customization note: For more information about the attributes and methods of the
-        TaskResult class, see lib/ansible/executor/task_result.py.
+        Customization note: Ansible uses only one console stdout
+        callback at a time. However, you can override this method in a
+        separate callback to add output to stdout or forward details
+        to a log, etc.
 
-        :param TaskResult result: An object that contains details about the task
+        :param result: An object that contains details about the task
+        :type result: TaskResult
 
         :return: None
         """
         if C.DISPLAY_SKIPPED_HOSTS:
             host = result._host.get_name()
             self.runner_on_skipped(host, self._get_item_label(getattr(result._result, 'results', {})))
 
-    def v2_runner_on_unreachable(self, result):
+    def v2_runner_on_unreachable(self, result: TaskResult) -> None:
+        """Process details of a task if a target node is unreachable.
+
+        Customization note: Ansible uses only one console stdout
+        callback at a time. However, you can override this method in a
+        separate callback to add output to stdout or forward details
+        to a log, etc.
+
+        :param result: An object that contains details about the task
+        :type result: TaskResult
+
+        :return: None
+        """
         host = result._host.get_name()
         self.runner_on_unreachable(host, result._result)
 
-    def v2_runner_on_async_poll(self, result):
+    def v2_runner_on_async_poll(self, result: TaskResult) -> None:
+        """Process details of a task running in asynchronous mode.
+
+        Customization note: Ansible uses only one console stdout
+        callback at a time. However, you can override this method in a
+        separate callback to add output to stdout or forward details
+        to a log, etc.
+
+        :param result: An object that contains details about the task
+        :type result: TaskResult
+
+        :return: None
```

Comment:
So this seems to be misused across the board. `:returns:` (`:return:`) is a text description of what a function returns.
There is a separate entry for the return type which would be
```suggestion
        :rtype: None
```

(though, they are usually used together)

Reply:
https://www.sphinx-doc.org/en/master/usage/domains/python.html#info-field-lists